keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -316,7 +322,7 @@ Additionally, privileged containers may impact other pod security policies (PSPs
    </td>
    <td>no
    </td>
-   <td>No support for pid and ipc since jobs don’t provide isolation at that level.
+   <td>HostPID and HostIPC will always be enabled because job objects do not support this level of isolation.
```

Comment:
if this is the case, should pod.spec.hostPID / hostIPC be required to be true as well for hostProcess pods by validation, to accurately reflect the shape of the resulting processes?

Reply:
Unlike with `hostNetwork` we have no plans to enable toggling `hostPID` or `hostIPC` in the future.
Because of that I was hoping to not require / validate this field and instead rely on documentation for the sake of usability / brevity in the deployment specs

Reply:
isn't it misleading to have hostPID/hostIPC be set to false when it is actually using the host pid/ipc space? a similar situation existed in the past with containerPorts when running in `hostNetwork: true` mode... all container ports in a pod are required to set hostPort == containerPort. Even though that's verbose, it tries to make the independent parts of the pod spec accurately reflect the implications of the `hostNetwork` field.

Reply:
I think those two settings are just oddballs in the Windows container world in general. Instead of having separate configurable namespaces like Linux all of the namespace concepts, besides networking which comes via the network namespace equivalent called network compartments, on Windows are all in one object called a silo. You're either in a silo and have all the isolation or none. Granted I have not read the K8s code handling these values on Windows but they shouldn't actually be changing any behavior on Windows as currently there's no way to achieve hostPID or hostIPC

Reply:
To me requiring hostPID/hostIPC for these containers seems a little bit like trying to fit a round peg (a Windows pod) into a square hole (current pod specs). If we were to split out security options for Windows pods in the future we wouldn't have hostPID/hostIPC fields because they don't really make sense for Windows containers but we would have a hostNetwork field (as Danny mentioned above Windows has an equivalent to network namespaces named network compartments)

Right now we just completely ignore these fields for Windows pods.
I'm concerned that if we started enforcing requiring these be set for `hostProcess` pods/containers users might get the impression that these would have some meaningful impact. Would we also add in validation to make sure hostPID and hostIPC are not set when the pod contains all normal containers?

In my opinion I think it would be better and more accurate to just heavily document the behaviors and capabilities of `hostProcess` containers.

I'm curious what @Random-Liu thinks here since he also commented on hostPID/hostIPC in this review.

Reply:
Also, I'm not sure how we would handle these fields if/when we support pods that can container both normal and `hostProcess` containers.
Since hostPID/hostIPC are at the pod level it would be misleading to set this to either true or false.
If  `hostNetwork=true` all of the containers would need to be `hostProcess` containers, but `hostNetwork=false` would be completely valid for a pod that contains both `hostProcess` and normal containers.

Reply:
> Also, I'm not sure how we would handle these fields if/when we support pods that can container both normal and `hostProcess` containers.
> Since hostPID/hostIPC are at the pod level it would be misleading to set this to either true or false.

Is the same true for the pod-level `hostNetwork` field in a future world where some containers can be `hostProcess:true` and some `hostProcess:false`, or would network visibility always be consistent pod-wide, even if some containers were not `hostProcess` containers?

> Granted I have not read the K8s code handling/validating these values on Windows but they shouldn't actually be changing any behavior on Windows

I'm mostly trying to avoid a future situation where windows runtimes gain the ability to control pid/ipc visibility and we're stuck with persisted values in existing pods that don't match the effective behavior of the containers. I'll defer to sig-node and sig-windows here to work this out, though. From a policy/security perspective, the `hostProcess` field will be much more significant to control.

Reply:
> Is the same true for the pod-level `hostNetwork` field in a future world where some containers can be `hostProcess:true` and some `hostProcess:false`, or would network visibility always be consistent pod-wide, even if some containers were not `hostProcess` containers?

Currently the plans to support a mix of `hostProcess` and normal containers in the same pod would enforce consistent pod-wide network visibility. (this would come in as another KEP, not an update between alpha/beta).
If `hostNetwork:true` then all containers in a pod would be required to set `hostProcess:true`.
If `hostNetwork:false` the pod could container both normal and `hostProcess` containers and all containers would have the same network visibility (pod network in this case).

> I'm mostly trying to avoid a future situation where windows runtimes gain the ability to control pid/ipc visibility and we're stuck with persisted values in existing pods that don't match the effective behavior of the containers.

We're actively working towards enabling the above mixed pod scenario (this requires updates at the Windows OS layer).
We don't have any plans to update/control pod/ipc visibility for Windows containers. 
This is a good thing to take into consideration but I think it will be very unlikely to ever be implemented for Windows containers.

Reply:
@liggitt I updated the table in the KEP to better reflect usage. PTAL.